### PR TITLE
Added popup to send an email to mobile users that join a collection for #3010

### PR DESCRIPTION
**Pull request by sylvieed**
_Resolves #3010_

### The problem

If a transcriber signs up to FromThePage on their phone and then wants to later come back to it on their computer, they have no easy way to find their account or the collection they joined again.

### The solution

This PR adds a popup to be shown after a user signs up through a collection that asks them if they'd like to be emailed a link to the collection. Here's how it works:

1. Someone (future user) gets referred to a collection's page on their mobile device
2. They sign up to FromThePage (and this generates a `collection_joined` deed)
3. Only if they are on a mobile device, a popup shows up (on the collection show page) saying that transcribing is easier on a larger screen, with a button to email them a link to the collection.
mobile device popup
4. If they click this button, they're sent the `new_mobile_user` email, which has a link to start transcribing that collection.
mobile device email
5. They're returned to the collection show page, with a green "Email sent." flash.

### Questions / Things to maybe change
* Right now, the popup only shows up if the user joined through a collection. Should it always come up?
* Would it be better to just always email users after they join a collection (#3011) instead of this mobile only & popup combo?
* I wasn't sure exactly what y'all wanted the popup & email to say. Let me know if you have any other wording in mind for these.

### Mobile layout issue

I tested this on my phone ([with ADB](https://stackoverflow.com/a/46795769/8326554)) and the popup actually looks really bad on mobile:
<p float="left" align="middle">
  Screenshot_20230122_214718_Chrome
  Screenshot_20230122_214725_Chrome
</p>

I'm not sure how to fix this because I don't really know how  `litebox`es work, it seems like all `litebox`es are doing this on mobile (ex: the API key popup looks very similar to this one above).

### CODE REVIEW

File: app/views/collection/show.html.slim [hunk 1]
```diff
@@ -138,3 +138,13 @@
...
+
+=link_to(collection_new_mobile_user_path(@collection.id), :data => { litebox: { hash: 'new-mobile-user' }}, id: 'new_mobile_user_link', style: "display: none")
+  span =""
+
+-if @new_mobile_user
+  -content_for :javascript
+    javascript: 
+      $(document).ready(function(){
+        $('#new_mobile_user_link').trigger('click');
+      });
```
Comment (sylvieed) [inline app/views/collection/show.html.slim:150]: This is how I got the popup to show up only when `@new_mobile_user` is true, but this code seems weird & bad practice to me. Do y'all have any insight into a way to do this better?

File: app/views/collection/new_mobile_user.html.slim [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.litebox-embed(style="width:550px")
```
Comment (sylvieed) [inline app/views/collection/new_mobile_user.html.slim:1]: Maybe this set `width` attribute is why the `litebox` looks weird on mobile

**Comment by sylvieed** [comment]
The tests are failing because of missing i18n translations, but I don't want to translate the new messages until we're sure about them, so we can ignore the failing tests for now.

File: app/views/collection/new_mobile_user.html.slim (hunk 2 above)
Comment (saracarl) [reply app/views/collection/new_mobile_user.html.slim:1]: Can we set the width narrower in this instance?

**Comment by saracarl** [comment]
I really like the pop up message, so let's keep that.
The mail message, though, could use more context.  Let's add the organization name (collection.owner) in the email "Milner Library's WWI Letters Collection" to remind them of what this is.  
Is there any way to sign it with the organization name like we're signing the notes emails (with a reply-to for the organization as well)?

While it would be lovely to do the custom welcome email.... hmmm....  I guess we could use the above as the default value and let customers change it in the collection settings tab.  It's just a text field there.  I'd be tempted to finish up this one, but then immediately go work on that one since it's in the same code and won't take as much re-familiarizing yourself.  

And if we read the mail content out of the DB we wouldn't have to translate it.  Project owners could if they wanted to.

**Review by saracarl — CHANGES_REQUESTED** [review]
See comment on PR

**Comment by sylvieed** [comment]
Changes made:
* The email is signed by the institution now
email

* The email subject was changed to "Owner's Collection name Collection" -- is this what you meant you wanted, or should the subject still have "Welcome to ..."? 
* I changed the width of the popup to `300px`, which looks much better in my mobile testing.

better popup
However, this isn't responsive, so it might look worse on other phones. It looks okay in landscape:

landscape

**Comment by saracarl** [comment]
@sylvieed I just tried this -- in order to take a screenshot -- and couldn't get it to work.  Any ideas why?  Want to try it yourself when you have time?

**Pull request merged**